core/src/mindustry/world/blocks/logic/CanvasBlock.java

```diff
@@ -214,7 +242,7 @@ public void draw(){
                 super.draw();
             }
 
-            if(texture == null){
+            if(texture == null || updated){
```

Comment:
I don't see you setting `updated` to `false` anywhere, meaning it will update every frame once it's set.

Reply:
> You set `update`, not `updated`

should be fixed in https://github.com/Anuken/Mindustry/pull/10899/commits/c4fab2d18e2463279cce034efae6bf64e948bb46